src/Modules/Reports/Commands.Reports/GetPowerBIImport.cs

```diff
@@ -0,0 +1,156 @@
+using Microsoft.PowerBI.Common.Abstractions.Interfaces;
+using Microsoft.PowerBI.Common.Api.Reports;
+using Microsoft.PowerBI.Common.Api.Workspaces;
+using Microsoft.PowerBI.Common.Client;
+
+namespace Microsoft.PowerBI.Commands.Reports
+{
+    [Cmdlet(CmdletVerb, CmdletName, DefaultParameterSetName = ListParameterSetName)]
+    [OutputType(typeof(IEnumerable<Import>))]
+    public class GetPowerBIImport : PowerBIClientCmdlet, IUserScope, IUserFilter, IUserId, IUserFirstSkip
+    {
+        public const string CmdletVerb = VerbsCommon.Get;
+        public const string CmdletName = "PowerBIImport";
+
+        #region ParameterSets
+        private const string IdParameterSetName = "Id";
+        private const string NameParameterSetName = "Name";
+        private const string ListParameterSetName = "List";
+        private const string ObjectIdParameterSetName = "ObjectAndId";
+        private const string ObjectNameParameterSetName = "ObjectAndName";
+        private const string ObjectListParameterSetName = "ObjectAndList";
+        #endregion
+
+        #region Parameters
+        [Alias("ImportId")]
+        [Parameter(Mandatory = true, ParameterSetName = IdParameterSetName)]
+        [Parameter(Mandatory = true, ParameterSetName = ObjectIdParameterSetName)]
+        public Guid Id { get; set; }
+
+        [Parameter(Mandatory = true, ParameterSetName = NameParameterSetName)]
+        [Parameter(Mandatory = true, ParameterSetName = ObjectNameParameterSetName)]
+        public string Name { get; set; }
+
+        [Parameter(Mandatory = false)]
+        public PowerBIUserScope Scope { get; set; } = PowerBIUserScope.Individual;
+
+        [Parameter(Mandatory = false, ParameterSetName = ListParameterSetName)]
+        [Parameter(Mandatory = false, ParameterSetName = ObjectListParameterSetName)]
+        public string Filter { get; set; }
+
+        [Alias("Top")]
+        [Parameter(Mandatory = false, ParameterSetName = ListParameterSetName)]
+        [Parameter(Mandatory = false, ParameterSetName = ObjectListParameterSetName)]
+        public int? First { get; set; }
+
+        [Parameter(Mandatory = false, ParameterSetName = ListParameterSetName)]
+        [Parameter(Mandatory = false, ParameterSetName = ObjectListParameterSetName)]
+        public int? Skip { get; set; }
+
+        [Alias("GroupId")]
+        [Parameter(Mandatory = false, ParameterSetName = IdParameterSetName)]
+        [Parameter(Mandatory = false, ParameterSetName = NameParameterSetName)]
+        [Parameter(Mandatory = false, ParameterSetName = ListParameterSetName)]
+        public Guid WorkspaceId { get; set; }
+
+        [Alias("Group")]
+        [Parameter(Mandatory = true, ParameterSetName = ObjectIdParameterSetName, ValueFromPipeline = true)]
+        [Parameter(Mandatory = true, ParameterSetName = ObjectNameParameterSetName, ValueFromPipeline = true)]
+        [Parameter(Mandatory = true, ParameterSetName = ObjectListParameterSetName, ValueFromPipeline = true)]
```

Comment:
It might be worthwhile to set ValueFromPipeline = true for the other cmdlets that have Workspace as a possible parameter. You could also consider renaming the ParameterSetName from WorkspaceParameterSetName to ObjectParameterSetName to make the existing cmdlets consistent.

Reply:
Somtimes I didn't want to set it to true if another ID was required as input